### PR TITLE
Migrate from alpha to beta compute APIs

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

The only Disk feature that required an alpha API (multiwriter) is now in beta.
We keep the infrastructure to be able to swap API versions, but it has been simplified to reduce code duplication.

**Special notes for your reviewer**:

Builds on top of #642

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/gce-cloud-provider/compute/fake-gce.go [hunk 1]
```diff
@@ -260,45 +260,32 @@ func (cloud *FakeCloudProvider) InsertDisk(ctx context.Context, volKey *meta.Key
 		}
 	}
 
-	var diskToCreate *CloudDisk
+	computeDisk := &computev1.Disk{
+		Name:             volKey.Name,
+		Zone:             volKey.Zone,
```
Comment (mattcary) [inline pkg/gce-cloud-provider/compute/fake-gce.go:265]: It looks like computeDisk.Zone is set in the meta.Zonal case below, and unused in the meta.Regional case. Maybe best not to set it here?

**Review by mattcary — COMMENTED** [review]
Thanks for cleaning up the disk organization, it was getting a little silly with all the types.

/lgtm

File: pkg/gce-cloud-provider/compute/fake-gce.go (hunk 1 above)
Comment (alculquicondor) [reply pkg/gce-cloud-provider/compute/fake-gce.go]: You are right. Removed

**Comment by alculquicondor** [comment]
/assign @msau42 
for approval

**Comment by mattcary** [comment]
/approve

oops

**Pull request merged**